plat/renesas/rcar_gen4/rcar_svc_setup.c

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2024 EPAM Systems
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include <common/debug.h>
+#include <common/runtime_svc.h>
+#include <drivers/scmi-msg.h>
+
+#include <rcar_smc.h>
+
+/* Setup Standard Services */
+static int32_t rcar_svc_setup(void)
+{
+	return 0;
+}
+
+/*
+ * Top-level Standard Service SMC handler.
+ */
+static uintptr_t rcar_svc_smc_handler(uint32_t smc_fid, u_register_t x1,
+				      u_register_t x2, u_register_t x3,
+				      u_register_t x4, void *cookie,
+				      void *handle, u_register_t flags)
+{
+	uint32_t ret1 = 0U;
+
+	switch (smc_fid) {
```

Comment:
can we avoid usage of switch case here:
```
 if (smc_fid >= RCAR_SIP_SMC_SCMI && smc_fid <= (RCAR_SIP_SMC_SCMI + 7 ) ) {
     scmi_smt_fastcall_smc_entry(smc_fid - RCAR_SIP_SMC_SCMI);
 } esle {
    WARN("Unknown RCAR Service Call: 0x%x\n", smc_fid);
    ret1 = SMC_UNK;
 }
```

Reply:
+1

Reply:
Rejected. It's beautification - I'm satisfied with switch.

Reply:
ok, than you can just reduce number of lines in this way:
```
	switch (smc_fid) {
	case RCAR_SIP_SMC_SCMI:
	case RCAR_SIP_SMC_SCMI + 1:
	case RCAR_SIP_SMC_SCMI + 2:
	case RCAR_SIP_SMC_SCMI + 3:
	case RCAR_SIP_SMC_SCMI + 4:
	case RCAR_SIP_SMC_SCMI + 5:
	case RCAR_SIP_SMC_SCMI + 6:
	case RCAR_SIP_SMC_SCMI + 7:
		scmi_smt_fastcall_smc_entry(smc_fid - RCAR_SIP_SMC_SCMI);
		break;
	default:
		WARN("Unknown RCAR Service Call: 0x%x\n", smc_fid);
		ret1 = SMC_UNK;
		break;
	}
```
or in this way:
```
	switch (smc_fid) {
	case RCAR_SIP_SMC_SCMI ... RCAR_SIP_SMC_SCMI + 7:
		scmi_smt_fastcall_smc_entry(smc_fid - RCAR_SIP_SMC_SCMI);
		break;
	default:
		WARN("Unknown RCAR Service Call: 0x%x\n", smc_fid);
		ret1 = SMC_UNK;
		break;
	}
```

Reply:
Rejected. No beautification changes will be made on this stage. Follow up patches are welcome.